test/integration/dialects/postgres/query-interface.test.js

```diff
@@ -6,15 +6,153 @@ var chai = require('chai')
   , Support = require(__dirname + '/../../support')
   , dialect = Support.getTestDialect()
   , DataTypes = require(__dirname + '/../../../../lib/data-types')
+  , Sequelize = require('../../../../index')
+  , Promise = Sequelize.Promise
   , _ = require('lodash');
 
+
 if (dialect.match(/^postgres/)) {
   describe('[POSTGRES Specific] QueryInterface', function () {
     beforeEach(function () {
       this.sequelize.options.quoteIdenifiers = true;
       this.queryInterface = this.sequelize.getQueryInterface();
     });
 
+    describe('createFunction', function () {
+
+      beforeEach( function () {
+        //make sure we don't have a pre-existing function called create_job
+        //this is needed to cover the edge case of afterEach not getting called because of an unexpected issue or stopage with the 
+        //test suite causing a failure of afterEach's cleanup to be called.
+        return this.queryInterface.dropFunction('create_job',[{type:'varchar',name:'test'}])
+        //suppress errors here. if create_job doesn't exist thats ok.
+        .catch( function (err) {  });
+      });
+
+      afterEach( function () {
+        //cleanup
+        return this.queryInterface.dropFunction('create_job',[{type:'varchar',name:'test'}])
+        //suppress errors here. if create_job doesn't exist thats ok.
+        .catch( function (err) {  });
+      });
+
+      it('creates a stored procedure', function () {
+        var self = this;
+        var body = 'return test;';
+        var options = {};
+
+        //make our call to create a function
+        return self.queryInterface.createFunction('create_job', [{type:'varchar',name:'test'}], 'varchar', 'plpgsql', body, options)
+        //validate
+        .then( function () { 
+          return self.sequelize.query('select create_job(\'test\');', { type: self.sequelize.QueryTypes.SELECT })
```

Comment:
Please chain the promises properly so it doesn't become a [promise hell](https://medium.com/@pyrolistical/how-to-get-out-of-promise-hell-8c20e0ab0513). Also get rid of `self` (where possible) by using either `bind` or `arrow functions` (preferred way). Same goes for other code section too